tf_agents/environments/rnd_wrapper.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+# Copyright 2018 The TF-Agents Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wrapper implementing Random Network Distillation (RND)."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import gin
+import gym
+import gym.spaces
+import numpy as np
+import tensorflow as tf
+
+from tf_agents import specs
+from tf_agents.environments import wrappers
+from tf_agents.networks import rnd_network
+from tf_agents.trajectories import time_step as ts
+from tf_agents.trajectories import trajectory
+from tensorflow.python.util import nest  # pylint:disable=g-direct-tensorflow-import  # TF internal
+
+
+@gin.configurable
+class RNDWrapper(wrappers.PyEnvironmentBaseWrapper):
+  def __init__(self, env):
+    super(RNDWrapper, self).__init__(env)
+
+    # TODO Freeze rnd_target_net
+    print('[rnd_wrapper.py] self.time_step_spec().observation: ', self.time_step_spec().observation)
+    print('[rnd_wrapper.py] type(self.time_step_spec().observation): ', type(self.time_step_spec().observation))
+
+    observation_spec = self.time_step_spec().observation
```

Comment:
If you keep the RND net in a wrapper you need to change your observation spec to have some structure so that you can expose the encoded features.

Otherwise the line below doesn't do anything.

Also don't think you need the redefining of the action_spec.